opennms-config/src/main/java/org/opennms/netmgt/config/service/ConfigurationService.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2002-2021 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2021 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.config.service;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+import java.util.Objects;
+
+import org.apache.commons.io.FileUtils;
+import org.opennms.core.utils.ConfigFileConstants;
+import org.opennms.core.xml.JaxbUtils;
+import org.springframework.stereotype.Service;
+
+
+/**
+ * We maintain and serve configurations from a centralized place in a uniform way.
+ */
+@Service
+public class ConfigurationService {
+
+    /**
+     * Loads the latest available configuration specified by the URI and transforms it into the given JaxB class.
+     */
+    public <T> T getConfigurationAsJaxb(final String uri, final Class<T> clazz) throws ConfigurationNotAvailableException {
```

Comment:
I think this could be a nice solution for getting specific configuration objects back without the service knowing about them. In a distributed system (configuration server lives in another JVM) this code would be executed in the client library that the internal user (e.g. vacuumd) calls.

Reply:
Makes sense.